Add preset net

Steffy-zxf opened the pull request:
add preset net (bilstm, bow, cnn, dpcnn, gru, lstm) for text classification task, which was only available as ERNIE/RoBERTa/BERT/ELECTRA module

Steffy-zxf committed:
add preset network (biltsm, bow, cnn, dpcnn, gru, lstm) for text classification task

Steffy-zxf committed:
Merge branch 'develop' of https://github.com/PaddlePaddle/PaddleHub into add-preset-net

Steffy-zxf committed:
add preset network

Steffy-zxf committed:
update preset net

Commit:
fix return_result bug

Commit:
update demo for preset net

Commit:
update preset usage

Commit:
update preset usage

ZeyuChen commented on demo/text_classification/predict.py:
```diff
@@ -34,6 +34,7 @@
 parser.add_argument("--max_seq_len", type=int, default=512, help="Number of words of the longest seqence.")
 parser.add_argument("--use_gpu", type=ast.literal_eval, default=False, help="Whether use GPU for finetuning, input should be True or False")
 parser.add_argument("--use_data_parallel", type=ast.literal_eval, default=False, help="Whether use data parallel.")
+parser.add_argument("--network", type=str, default='bilstm', help="Preset network which was connected after Transformer model, such as ERNIE, BERT ,RoBERTa and ELECTRA.")
```
预置网络，Predefined networks

Steffy-zxf replied:
> 预置网络，Predefined networks

已修改

ZeyuChen commented on paddlehub/finetune/task/classifier_task.py:
```diff
@@ -156,21 +161,73 @@ def _postprocessing(self, run_states):
                  feed_list,
                  data_reader,
+                 token_feature=None,
+                 feature=None,
```
feature 要定义在在 token_feature前面比较合理。

Steffy-zxf replied:
已修改

Steffy-zxf replied:
> feature 要定义在在 token_feature前面比较合理。

已修改

ZeyuChen commented on paddlehub/finetune/task/classifier_task.py:
```diff
@@ -156,21 +161,73 @@ def _postprocessing(self, run_states):
+            feed_list(list): the variable name that will be feeded to the main program
+            data_reader(object): data reader for the task. It must be one of ClassifyReader and LACClassifyReader.
+            token_feature(Variable): the feature will be used to connect the preset net. It must be the token-level feature, shape as [-1, seq_len, emb_size]. Default None.
+            feature(Variable): the feature will be used to classify texts. It must be the sentence-level feature, shape as [-1, emb_size]. Token_feature and feature couldn't be setted as the same time. One of them must be setted as not None. Default None.
```
Token_feature and feature couldn't be->
`token_feature` and `feature`
涉及参数变量需要用``括起来，避免歧义。

Steffy-zxf replied:
> Token_feature and feature couldn't be->
> `token_feature` and `feature`
> 涉及参数变量需要用``括起来，避免歧义。

已修改

ZeyuChen commented on paddlehub/finetune/task/classifier_task.py:
```diff
@@ -179,10 +236,33 @@ def __init__(self,
+                self.feature, length=self.seq_len_used)
+
+        if self.network:
+            # add preset net
```
pre-defined

Steffy-zxf replied:
> pre-defined

done

ZeyuChen commented on paddlehub/finetune/task/classifier_task.py:
```diff
@@ -179,10 +236,33 @@ def __init__(self,
+                "%s has been added in the TextClassifierTask!" % self.network)
+        else:
+            # not use preset net but to use fc net
+            cls_feats = fluid.layers.dropout(
```
This is dropout, where's the FC layer?

Steffy-zxf replied:
> This is dropout, where's the FC layer?

line 267-280

ZeyuChen commented on paddlehub/network/classification.py:
```diff
@@ -0,0 +1,139 @@
+This module provide nets for text classification
+"""
+
+from paddlehub.common.utils import version_compare
```
It seems you can remove this import for simplify header

Steffy-zxf replied:
> It seems you can remove this import for simplify header

done

ZeyuChen commented on paddlehub/reader/nlp_reader.py:
```diff
@@ -293,28 +294,28 @@ def _pad_batch_records(self, batch_records, phase=None):
 
             return_list = [
```
`return_list`代码有点冗余。
是否可以通过if else的判断后，在原来return_list的基础上进行append，而不用每次都进行重新定义，这样代码也比较冗长。

Steffy-zxf replied:
> `return_list`代码有点冗余。
> 是否可以通过if else的判断后，在原来return_list的基础上进行append，而不用每次都进行重新定义，这样代码也比较冗长。

已修改

ZeyuChen reviewed (changes requested):
Please update your code according to my comments.

Commit:
1.update nlp_reader, remove redundant returns

2. fix pre-set net spelling , pre-set net -> pre-defined net

3. remove version_compare usage

Commit:
fix preset net spelling, preset net -> pre-defined net

Commit:
fix typo

Commit:
add predefine network usage demo

Commit:
add predefine network usage demo

ZeyuChen reviewed (approved):
LGTM
